freqtrade/freqai/utils.py

```diff
@@ -233,3 +233,23 @@ def get_timerange_backtest_live_models(config: Config) -> str:
     dd = FreqaiDataDrawer(models_path, config)
     timerange = dd.get_timerange_from_live_historic_predictions()
     return timerange.timerange_str
+
+
+def ensure_base_tf_in_include_timeframes(config: Config) -> Config:
+    """
+    Ensure that the base timeframe is included in the include_timeframes list
+    :param config: Configuration dictionary
+
+    :return config: Configuration dictionary
+    """
+    feature_parameters = config.get('freqai', {}).get('feature_parameters', {})
+    include_timeframes = feature_parameters.get('include_timeframes', [])
+
+    if config['timeframe'] in include_timeframes:
+        return config
+
+    include_timeframes = [config['timeframe']] + include_timeframes
```

Comment:
> Why is this a requirement?
> 
> i think i should be fine running on 5m data (main tf) - but have `"include_timeframes": ["15m", "30m", "1h"]`.
> 
> The include_timeframes can't be smaller than the main timeframe (so changing main to 30m won't work - but this is validated) - but i'm not aware why my above setup wouldn't be valid ?

It is not valid, currently including base timeframe in include_parameters is a requirement. But there is no validation. You just get KeyError if you try to run without including base timeframe in your include_timeframes config. 

This function will allow users to forget base timeframe in include_timeframes. 

I could do this without modifying config here. But there were a lot of loops for getting data and other stuff so this was an easy method to do it. 

If this doesn’t look ideal to you we can of course change all loops of include_timeframes one by one in all files.

Reply:
interesting, seems like i never ran into this so far, but i thought i had tested exactly this.

To be clear - I'm not questioning the method (though you might want to do this in `_validate_freqai_include_timeframes` - so all related logic is in one place) - just the requirement - as it's new to me.

Reply:
You mean updating config in that function or throwing an error if base timeframe not in include_timeframes?

I think since those validation functions doesn’t return anything updating config inside there is not possible without a change?

Reply:
By the way we can still remove this requirement by changing all loops of include_timeframes.

Reply:
`dict.update()` will mutate the original dictionary. 
Otherwise your method would break tons of stuff, as then we'd have 2 configurations around - one you modified, and one that was there before.

> By the way we can still remove this requirement by changing all loops of include_timeframes.

i'm still not a fan of this ... :laughing:

Reply:
> `dict.update()` will mutate the original dictionary.

Ok, I will move the logic there. Thanks for the info.